### ISSUE TITLE
Hedera-side fees per user

### ISSUE THREAD

**Issue by vectorman1** [LimeChain/hashport-validator#682]
TLDR: 

- Use ~HFS~/HCS for storing the fee policies for all Hedera native-related operations
    - YAML for fee policies → submit in the topic;
    - validators listen to the topic + chunk logic
    - validators load the YAML in memory
    - similar to bridge config
    - HBAR → EVM; HTS → EVM
    - sender account is used for identification of the fee policy

HFS ruled out for storing fee policies as there is no way to get file contents from the mirror-node, which means it would cost network fees to read files. 

<hr>

1. Proposed fee policy model
```yaml
type: "flat | percentage | tiered_percentage" 
amount: 10000 # in case of flat fee
percentage: 500 # in case of percentage fee
tiers: # in case of tiered fee
   - lte: 1000 # less-than-or-equal range of amount of token where the tier is applied
     percentage: 20000 # 20.00%
   - lte: 10000 
     percentage: 10000 # 10.00%
```

2. Create a topic that will contain the fee policies for all clients requiring them.
Proposed YAML structure: 
```yaml
policies:
  <chain ID>: 
    <token ID | address>:
       <sender acc ID | address>: 
          type: flat
          amount: 500  
```
 The policies should be sent to the topic in chunks utilizing [ConsensusMessageChunkInfo](https://docs.hedera.com/guides/docs/hedera-api/consensus-service/consensussubmitmessage#consensusmessagechunkinfo), assembled on start of the validator and loaded in memory. 

⚠️ It is important to validate the policies, e.g. 
 - NFT assets cannot have a percentage or tiered base fee policy. 
 - There should be only 1 policy per asset per account. 

3. Whenever a transfer is initiated, look up this fee policy configuration for an alternate fee policy for the originator account ID of the transaction. If such exists, use it instead.

[User Stories](https://www.notion.so/limechain/Fee-Per-User-User-Stories-f7fc1ebd1cdc45a98c8f50b74ecc968f)

**Comment by Daniel-K-Ivanov** [comment]
```yaml
policies:
  <chain ID>: 
    <token ID | address>:
       <sender acc ID | address>: 
          type: flat
          amount: 500  
```

We have to support 1 entity having more than 1 address that can utilise the fee policy. The other thing is that all addresses for a given entity will have the same policy.

**Comment by bibitibooo1** [comment]
[Notion file](https://limechain.notion.site/limechain/MetaLock-Notes-Potentially-change-the-name-056ff5f92ce6459ba6721ab0c3c29ed4)
